taichi/program/async_engine.cpp

```diff
@@ -264,50 +264,47 @@ TaskMeta AsyncEngine::create_task_meta(const TaskLaunchRecord &t) {
     if (auto global_load = stmt->cast<GlobalLoadStmt>()) {
       if (auto ptr = global_load->ptr->cast<GlobalPtrStmt>()) {
         for (auto &snode : ptr->snodes.data) {
-          meta.input_states.emplace_back(snode, AsyncState::Type::value);
+          meta.input_states.insert({snode, AsyncState::Type::value});
         }
       }
     }
     if (auto global_store = stmt->cast<GlobalStoreStmt>()) {
       if (auto ptr = global_store->ptr->cast<GlobalPtrStmt>()) {
         for (auto &snode : ptr->snodes.data) {
-          meta.output_states.emplace_back(snode, AsyncState::Type::value);
-          if (ptr->activate)
-            meta.output_states.emplace_back(snode, AsyncState::Type::mask);
+          meta.input_states.emplace(snode, AsyncState::Type::value);
```

Comment:
Why do we have `input_states` here? It looks like mixing things like RAW and WAW up.